exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -296,6 +303,7 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 
 				// Continue without adding the error to the aggregate if we can't find the resource.
 				if apierrors.IsNotFound(err) {
+					resourceNotFound = true
```

Comment:
I had to change the logic here to keep the unit tests green.

Basically before the change we ignored not found errors. This means when a new ConfigMap that is part of a ClusterResourceSet is created we get an event, but we can hit not found here if the cache is not up to date yet.

As we just ignored the error and didn't requeue we would have to wait a full resync period to reconcile the ConfigMap.
WIth this change we requeue now and the ConfigMap is reconciled quickly.

I have no idea what triggered this compared to the old CR version, but also no idea how to figure out what the difference is as I cannot reproduce it locally.

Reply:
Moved to: https://github.com/kubernetes-sigs/cluster-api/pull/8611